tests/mir-opt/const_prop/offset_of.rs

```diff
@@ -0,0 +1,25 @@
+// unit-test
+// compile-flags: -O
+
+// EMIT_MIR offset_of.main.ConstProp.diff
+
+#![feature(offset_of)]
+
+use std::mem::offset_of;
+
+#[repr(C)]
+struct Foo {
+    x: u8,
+    y: u16,
+    z: Bar,
+}
+
+#[repr(C)]
+struct Bar(u8, u8);
+
+fn main() {
+    let x = offset_of!(Foo, x);
+    let y = offset_of!(Foo, y);
+    let z0 = offset_of!(Foo, z.0);
+    let z1 = offset_of!(Foo, z.1);
+}
```

Comment:
Could you also add the test with a partially generic with known layout, and one on a partially generic with unknown layout?